bundles/org.openhab.binding.smaenergymeter/src/main/java/org/openhab/binding/smaenergymeter/internal/packet/DefaultPacketListenerRegistry.java

```diff
@@ -68,8 +67,8 @@ protected void shutdown() throws IOException {
         scheduler.shutdownNow();
     }
 
-    public ScheduledFuture<?> addTask(Runnable runnable, int intervalSec) {
-        return scheduler.scheduleWithFixedDelay(runnable, 0, intervalSec, TimeUnit.SECONDS);
+    public ScheduledFuture<?> addTask(ReceivingTask runnable) {
+        return scheduler.scheduleWithFixedDelay(runnable, 0, 1000, TimeUnit.MILLISECONDS);
```

Comment:
I'm not familair with this device, but a 1 second interval seems a little short, if i'm correct, previously this was 30 seconds.

Reply:
Issue is - device sends packets every second, and we received it only at specified interval. This change allows us to catch all packets as they arrive to avoid queuing them by operating system network layer. Filtering and handling of throttling is moved above packet listener itself.
I'm awaiting a test run from reporter (@cambronf) to see if this change helps.

Reply:
@monnimeter are you able to test this or comment?

Reply:
@lsiepel I've ran sma meter simulator and found one more leak which is fixed in most recent code.